types/underscore/index.d.ts

```diff
@@ -4621,8 +4641,24 @@ declare module _ {
          * the second elements, and so on. (alias: transpose)
          * @returns The unzipped version of the wrapped lists.
          **/
-        unzip(): any[][];
-        transpose(): any[][];
+        unzip(): T extends [infer A, infer B, infer C]
+            ? [A[], B[], C[]]
+            : T extends [infer A, infer B]
+            ? [A[], B[]]
+            : T extends [infer A] // eslint-disable-line no-single-element-tuple-type
+            ? [A[]] // eslint-disable-line no-single-element-tuple-type
+            : T extends List<infer A>
+            ? A[][]
+            : never[];
+        transpose(): T extends [infer A, infer B, infer C]
+            ? [A[], B[], C[]]
+            : T extends [infer A, infer B]
+            ? [A[], B[]]
+            : T extends [infer A] // eslint-disable-line no-single-element-tuple-type
+            ? [A[]] // eslint-disable-line no-single-element-tuple-type
+            : T extends List<infer A>
+            ? A[][]
+            : never[];
```

Comment:
If I understand correctly, this is a destructive change; unlike the other changes, there is no fallback to `any[][]`.